src/dual_context.jl

```diff
@@ -58,15 +58,13 @@ end
         vals, pushfwd = res
         ps = _partialss(tag, args)
 
-        if !(pushfwd isa Tuple)
-            # a single function (scalar output)
-            d = overdub(ctx, pushfwd, Zero(), ps...)
-            return Dual{T}(vals, d)
-        else
-            # many partial functions (as many as outputs)
-            return map(vals, pushfwd) do val, ∂
-                Dual{T}(val, overdub(ctx, ∂, Zero(), ps...))
+        ∂s = overdub(ctx, pushfwd, Zero(), ps...)
+        return if ∂s isa Tuple
+            map(vals, ∂s) do val, ∂
+                Dual{T}(val, ∂)
```

Comment:
Why not `Dual{T}.(vals, ∂s)` ?
rather than using `map` ?

Reply:
That introduces unnecessary work for the compiler

Reply:
I take it you've tested this and timed it?
cos it is more readable.

Reply:
I don't see how the broadcast is more readable than `map`. One shouldn't use broadcast is most cases. `map` is more readable and more robust because you know the args must all have the same length.

map:
```julia
 194.01 M allocations: 10.253 GiB, 4.04% gc time
```

broadcast:
```julia
195.45 M allocations: 10.334 GiB, 4.41% gc time
```

Reply:
It's readable enough to cut down on compile times. We will really need to watch compile times with this library, so we should cut every corner we can.